libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -360,7 +360,16 @@ public IEnumerable<Dialog> GetDependencies()
         {
             EnsureDependenciesInstalled();
 
-            yield break;
+            foreach (var trigger in Triggers)
+            {
+                if (trigger is IDialogDependencies depends)
+                {
+                    foreach (var dlg in depends.GetDependencies())
+                    {
+                        yield return dlg;
+                    }
+                }
+            }
```

Comment:
this doesn't seem right.  This will register ALL actions all the way up to the root.  Don't you want a filter on 
```
   foreach(var dlg in depends.GetDependencncies().Where(dlg => dlg is DialogContainer)) 
```

Reply:
Also, can you add a unit test to verify that only DialogContainers get automatically added to the DialogManager.Dialogs collection?

---
In reply to: [510264709](https://github.com/microsoft/botbuilder-dotnet/pull/4839#discussion_r510264709) [](ancestors = 510264709)

Reply:
So this is not about the registration, in fact this is unrelated to the dialog manager change. `GetDependencies` is supposed to return the dialogs on which the `IDialogDependencies` depends on. However the current implementation returned nothing. For example, the `IfCondition` action implements `IDialogDependencies` and returns the if and else dialogs in the implementation. This method should return whatever dialogs the adaptive dialog depends upon.

Now the next part is: The Dialog manager needs to deal with Containers and register only those. So to that end, the dialog manager has a filter on dependencies for container types only.

All that said, I don't really need this for the Adapters change, but felt like we are not implementing the `IDialogDependencies` interface correctly. I can remove this part and all will still work.

Reply:
Adding the test in the next iteration.

Reply:
Tom and I chatted offline, basically AdaptiveDialog implements IDialogdependencies but it has no public dependencies. Removing this part :)